test/lit/passes/signature-refining.wast

```diff
@@ -985,3 +985,60 @@
   (struct.new $C) ;; this will allow this function's result to be refined to $C
  )
 )
+
+;; Test we consider call.without.effects when deciding what to refine. $A has
+;; two subtypes, B1 and B2, and a call.without.effects sends in one while a
+;; normal call sends in the other. As a result, we cannot refine to either.
```

Comment:
It would be nice to additionally test that the refinement can still happen if there is a call without effects involved and the types work out.

Reply:
Sounds good. I'll start automerge now as I am gone for the evening and want to get this on the way for the user that hit it, and will open a followup PR tomorrow with that extra test.